store/src/java/com/zimbra/cs/filter/jsieve/EditHeaderExtension.java

```diff
@@ -564,4 +566,35 @@ private boolean matchValue(String regex, String value) {
         Matcher matcher = pattern.matcher(value);
         return matcher.matches();
     }
+
+    static public boolean isTolerableFormedMessage(ZimbraMailAdapter zma, String actionName, MimeMessage mm) {
+        if (zma.getEheParseStatus() == PARSESTATUS.UNKNOWN) {
+            return saveChanges(zma, actionName, mm);
+        } else if (zma.getEheParseStatus() == PARSESTATUS.MIMEMALFORMED) {
+            ZimbraLog.filter.info(actionName + ": Triggering message is malformed MIME");
+            return false;
+        }
+        return true;
+    }
+
+    static public boolean saveChanges(ZimbraMailAdapter zma, String actionName, MimeMessage mm) {
+        String ct = "";
+        try {
+            ct = mm.getContentType();
+        } catch (MessagingException e) {
+            ct = "(unknown)";
+        }
+
+        try {
+            mm.saveChanges();
+        } catch (ParseException pe) {
```

Comment:
No, in this context, ParseException is the one which I want to treat as a non-error status.  In other words, the EditHeaderException.saveChanges() should pretend to be fine (and return true), if and only if the ParseException occurs; because the it was caused by the formatting error found in the unrelated headers.